Assignment4/KairoSagewing5-22a292/task07.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1KDva5GdAK3jj10OKPKuf4i20LtKwxmfl
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+for s, p, o in g:
+  print (s, p, o)
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+
+NS = Namespace("http://somewhere#")
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT ?Sub ?SubSub WHERE { 
+    ?Sub RDFS:subClassOf ns:Person.
+    OPTIONAL {?SubSub RDFS:subClassOf ?Sub.}
+  }
+  ''',
+  initNs = { "ns": NS, "RDFS" : RDFS}
+)
+
+# Visualize the results
+
+for r in g.query(q1):
+  print(r.Sub, r.SubSub)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+q2 = prepareQuery('''
+  SELECT ?Persons WHERE { 
+    {
+      ?Persons RDF:type ns:Person.
+    }
+    UNION
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+      ?Persons RDF:type ?Sub.
+    }
+    UNION
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+        OPTIONAL {?SubSub RDFS:subClassOf ?Sub.}.
+      ?Persons RDF:type ?SubSub.
```

Comment:
This does not cover more than 3 levels in the taxonomy. Use * to chain subClassOf properties and retrieve all of them